ngcplogger.go

```diff
@@ -345,6 +362,97 @@ func (l *nGCPLogger) extractMsgFromPayload(m map[string]any) {
 	}
 }
 
+func (l *nGCPLogger) extractGcpFromPayload(m map[string]any, entry *logging.Entry) {
+
+	if l.extractGcp {
+		if val, exists := m["logging.googleapis.com/sourceLocation"]; exists {
+			v := val.(map[string]any)
+			entry.SourceLocation = &loggingpb.LogEntrySourceLocation{
+				File:     v["file"].(string),
+				Line:     int64(v["line"].(float64)),
+				Function: v["function"].(string),
+			}
+			delete(m, "logging.googleapis.com/sourceLocation")
+		}
+		if val, exists := m["logging.googleapis.com/trace"]; exists {
+			entry.Trace = val.(string)
+			delete(m, "logging.googleapis.com/trace")
+		}
+		if val, exists := m["logging.googleapis.com/spanId"]; exists {
+			entry.SpanID = val.(string)
+			delete(m, "logging.googleapis.com/spanId")
+		}
+		if val, exists := m["logging.googleapis.com/trace_sampled"]; exists {
+			entry.TraceSampled = val.(bool)
+			delete(m, "logging.googleapis.com/trace_sampled")
+		}
+		if val, exists := m["logging.googleapis.com/labels"]; exists {
+			v := val.(map[string]any)
+			for k, v := range v {
+				entry.Labels[k] = v.(string)
+			}
+			delete(m, "logging.googleapis.com/labels")
+		}
+	}
+}
```

Comment:
All the GCP functionality is already present in MR #9, it might cause conflicts when that one is merged first, and ideally only the changes related to caddy and http request info should be included in this PR, as it is titled that way. Just FYI :)

Reply:
I will delete this pull request and resubmit with the requested changes and any conflicts resolved later today, I will probably have some time sitting and waiting in the airport.